src/text-buffer.js

```diff
@@ -1927,16 +1927,29 @@ class TextBuffer {
       try {
         await this.buffer.save(destination, this.getEncoding())
       } catch (error) {
-        const canEscalate = process.platform === 'darwin' || process.platform === 'linux'
-        if (error.code === 'EACCES' && destination === filePath && canEscalate) {
+        if (error.code !== 'EACCES' || destination !== filePath) throw error
+
+        const isMacOrLinux = process.platform === 'darwin' || process.platform === 'linux'
+        const isWindows = process.platform === 'win32'
+        if (isMacOrLinux) {
           const fsAdmin = require('fs-admin')
           try {
             await this.buffer.save(fsAdmin.createWriteStream(filePath), this.getEncoding())
           } catch (_) {
             throw error
           }
-        } else {
-          throw error
+        } else if (isWindows) {
+          const winattr = require('winattr')
```

Comment:
You can cache `winattr` to speed up the execution. Just define `winattr` above the file:
```
let winattr = null
```

```suggestion
		  if (!winattr) {
		    winattr = require('winattr')
		  }
```

Reply:
@gosukiwi Check here. I have promisified the functions. I have submitted a PR to the original repository:
https://github.com/stevenvachon/winattr/pull/12

We can either wait for that, or we promisify the functions ourselves.

Reply:
For that I copied the way the original code does it:

```javascript
if (isMacOrLinux) {
  const fsAdmin = require('fs-admin')
  // ...
} else (isWindows) {
  const winattr = require('winattr')
}
```

I like the fact that it's lazy and wont be required unless needed, besides, node will cache it later.

Reply:
We need to cache the promisified functions though.

Reply:
Does Atom provide promisify? I don't think introducing the `promisify` dependency is a good idea otherwise. Besides, what advantages are we really getting with it? I'd rather have simple code, and measure before trying to optimize.

Reply:
Ohh does `promisify` lives in `util`?